src/pool-cl/libraries/CLPool.sol

```diff
@@ -408,9 +415,13 @@ library CLPool {
 
         ///@dev update user position and collect fees
         /// must be done after ticks are updated in case of a 0 -> 1 flip
-        (cache.feesOwed0, cache.feesOwed1) = self.positions.get(params.owner, params.tickLower, params.tickUpper).update(
-            params.liquidityDelta, cache.feeGrowthInside0X128, cache.feeGrowthInside1X128
-        );
+        CLPosition.Info storage position = self.positions.get(params.owner, params.tickLower, params.tickUpper);
+        if (position.liquidity == 0 && position.feeGrowthInside0LastX128 == 0 && position.feeGrowthInside1LastX128 == 0)
+        {
+            self.positionKeys.push(PositionKey(params.owner, params.tickLower, params.tickUpper));
```

Comment:
should we consider doing this off-chain ?

Reply:
on the same page, why are we doing this on-chain? the PR description doesn't explain this

Reply:
got it, let me explain it in today's v4 sync